### ISSUE TITLE
Request to Modify Authorization Header Handling for Specific Endpoints

### ISSUE THREAD

**Issue by MuhammadFaizanHaidar1** [TangibleInc/template-system#142]
Hi [Loops and logic Developer's],

I hope this message finds you well.

We’re currently using the [CheckView] to expose custom REST API endpoints on our WordPress site. Our plugin relies on JWT tokens passed in the Authorization header to authenticate API calls.

However, we’ve encountered a conflict between our plugin and Loops and Logic because of how the Authorization header is being processed globally. Specifically, your plugin appears to handle all REST API requests with the following code snippet:

add_filter( 'rest_pre_dispatch', array( $this, 'rest_pre_dispatch' ), 10, 3 );

$header_name = 'HTTP_AUTHORIZATION';
$auth        = isset( $_SERVER[ $header_name ] ) ? $_SERVER[ $header_name ] : false;

if ( ! $auth ) {
    $auth = isset( $_SERVER['REDIRECT_HTTP_AUTHORIZATION'] )
        ? $_SERVER['REDIRECT_HTTP_AUTHORIZATION']
        : false;
}
This code is attempting to validate the Authorization header for every REST API request, including those unrelated to the Loops and Logic API. Since our plugin also uses the Authorization header for JWT-based validation, this results in errors when the Loops and Logic plugin tries to interpret our JWT tokens as API keys.

Suggested Solution
To prevent conflicts and ensure both plugins can coexist without issues, we kindly request that you modify the header validation logic to only process Authorization headers for your specific endpoints.

Here is an example of how this can be achieved:

$requested_route = rest_get_url_prefix() . '/your-api-base'; // Replace with your API base route.if ( strpos( $_SERVER['REQUEST_URI'], $requested_route ) !== false ) {
    $header_name = 'HTTP_AUTHORIZATION';
    $auth        = isset( $_SERVER[ $header_name ] ) ? $_SERVER[ $header_name ] : false;

    if ( ! $auth ) {
        $auth = isset( $_SERVER['REDIRECT_HTTP_AUTHORIZATION'] )
            ? $_SERVER['REDIRECT_HTTP_AUTHORIZATION']
            : false;
    }
    
    // Proceed with your plugin's authentication logic here.
}
Thanks!

**Comment by zackpyle** [comment]
@eliot-akiEl This is causing an issue for me where I have L&L and CheckView on a few sites. Can you please look into it? Thanks!

**Comment by eliot-akira** [comment]
Hi @MuhammadFaizanHaidar1 - Thanks for the help in solving this plugin compatibility issue.

I found that the proposed change to the API request handler causes some of our existing tests to fail because the JWT token validation is expected to happen for _all_ REST endpoints, not just under the namespace `tangible`. This is necessary to provide the current user if there is a valid token. I believe it's the same for all plugins that implement JWT authentication, including CheckView.

So the solution is to let both plugins do this without interfering with each other.

In your pull request #144, the filter `rest_pre_dispatch` would have returned `false` for routes other than our specific namespace. https://github.com/TangibleInc/template-system/pull/144/files#diff-a67a238cb8621f4efe2ebbf3525983a7712a488fff582934bced5d59aba34473R491

However, the filter is expected to return `null` to not override the request, as documented here: https://developer.wordpress.org/reference/hooks/rest_pre_dispatch/#parameters

In a new commit 99c3cf5, I improved the logic so that if the parameter `$result` is not empty - meaning another filter returned a value to override the response already - it skips the token validation. Also, even if the validation fails or throws an error, it always returns `null` by default.

Would you mind confirming that this solves the compatibility issue with CheckView?

Here is the beta release at the newest commit. https://github.com/TangibleInc/template-system/releases/download/latest/tangible-template-system-latest.zip

**Comment by eliot-akira** [comment]
@zackpyle Thanks for the feedback. There's been some larger updates in the plugin framework, including a more thorough build and test workflow. I've also made an update to address the error mentioned.

Could you try [the latest version of the template system](https://github.com/TangibleInc/template-system/releases/download/latest/tangible-template-system-latest.zip) and see if the original issue has been resolved?

**Comment by zackpyle** [comment]
Yup, all working now. And Checkview now works when that version is activated 👍🏻

**Comment by MuhammadFaizanHaidar1** [comment]
Glad this has been resolved, thank you both.

Matt Schwartz

Director of Client Services

WordPress and Certified WooExperts

**Comment by eliot-akira** [comment]
Thanks for confirming that it works. It took longer than expected, but I hope to get this released in the next version and will let you know.

**Comment by eliot-akira** [comment]
This has been resolved in L&L 4.1.8.

**Comment by MuhammadFaizanHaidar1** [comment]
Great, thank you again.

Matt Schwartz

Director of Client Services

WordPress and Certified WooExperts
